sdk/storage/azure-storage-blob/samples/client_side_encryption_keyvault.py

```diff
@@ -0,0 +1,118 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: client_side_encryption_keyvault.py
+
+DESCRIPTION:
+    This sample contains code demonstrating how to conifgure the storage blob
+    service for client side encryption, storing and retrieving the key encryption key
+    (kek) from within Azure KeyVault. This sample requires a service principal be set
+    configured with access to KeyVault, and that the vault contains a 256-bit base64-
+    encoded secret named "symmetric-key". Additionally, a number of environment
+    variables, listed below, must be set. Since these often contain sensitive information,
+    they SHOULD NOT be replaced with hardcoded values in any code derived from this sample.
+
+USAGE: python client_side_encryption_keyvault.py
+"""
+
+import base64
+import os
+import sys
+import uuid
+
+from azure.identity import ClientSecretCredential
+
+from azure.keyvault.keys.crypto import CryptographyClient, KeyWrapAlgorithm
+from azure.keyvault.keys import KeyVaultKey, KeyType
+from azure.keyvault.secrets import SecretClient
+
+from azure.storage.blob import BlobServiceClient, BlobType
+
+# Environment variable keys which must be set to run this sample
+STORAGE_URL = 'AZURE_STORAGE_URL'
```

Comment:
What's it called in the portal, "blob service endpoint"?

Reply:
Blob endpoint. Renamed to `AZURE_BLOB_ENDPOINT` to mesh with the Python guidelines on env var names.

Reply:
Also, updated KEYVAULT_URL to KEYVAULT_DNS_NAME to more closely match the portal.

Reply:
Can we keep KEYVAULT_URL? Key Vault libraries have aligned on `vault_url`.